src/Interpreters/GinFilter.cpp

```diff
@@ -146,14 +146,23 @@ bool matchAnyInRange(const GinPostingsCache & postings_cache, UInt32 segment_id,
         /// Check if it is in the same segment by searching for segment_id
         const GinSegmentedPostingsListContainer & container = term_postings.second;
         if (auto container_it = container.find(segment_id); container_it != container.cend())
-            postings_bitset |= *container_it->second;
+        {
+            const GinIndexPostingsList & segment_posting_bitset = *container_it->second;
+            const auto minimum_in_list = segment_posting_bitset.minimum();
+            const auto maximum_in_list = segment_posting_bitset.maximum();
+
+            /// Check if the postings list has always match flag
```

Comment:
I am not sure I understand the PR.
Does "always match flag" (l. 154) correspond to "minimum_in_list == UINT32_MAX"? Maybe we can pull the condition somehow out into a variable or function which we can check/call here.